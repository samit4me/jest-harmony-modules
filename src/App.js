import React, { Component } from 'react';
import classNames from 'classnames';
import { square } from '000';
import logo from './logo.svg';
import styles from './App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.squared = this.squared.bind(this);
  }

  squared(x) {
    const test = 'jjjjjjj';
    return square(`etff_${test}`, styles);
  }

  render() {
    const style = this.squared(2);
    const className = classNames(style, styles.stuff);
    return (
      <div className={className}>
        <div className={styles['App-header']}>
          <img src={logo} className={styles['App-logo']} alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <p className={styles['App-intro']}>
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <p>square(2) === {this.squared(2)}</p>
      </div>
    );
  }
}

export default App;
