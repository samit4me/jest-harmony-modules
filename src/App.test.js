import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

it('renders without crashing', () => {
  shallow(<App />);
});

it('squared method works correctly', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.instance().squared(2)).toBe(4);
});