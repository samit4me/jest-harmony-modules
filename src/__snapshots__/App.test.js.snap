exports[`test renders correctly 1`] = `
<div
  className="etff_jjjjjjj stuff">
  <div
    className="App-header">
    <img
      alt="logo"
      className="App-logo"
      src="test-file-stub" />
    <h2>
      Welcome to React
    </h2>
  </div>
  <p
    className="App-intro">
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <p>
    square(2) === 
    etff_jjjjjjj
  </p>
</div>
`;
